### PR TITLE
Move some stdlib tests from `tests/ui` to `library/std/tests`

### PR DESCRIPTION
**Pull request by rytheo**
Related to #99417

### CODE REVIEW

File: library/std/tests/issue-21058.rs
Comment (workingjubilee) [inline library/std/tests/issue-21058.rs:1]: Please try to give an actual name to this test. What is it testing?

**Review by workingjubilee — COMMENTED** [review]
Thank you for the cleanup. One request:

File: library/std/tests/issue-21058.rs
Comment (rytheo) [reply library/std/tests/issue-21058.rs:1]: I've updated the name and moved the description into the function's doc comment.

**Comment by workingjubilee** [comment]
Thank you!

@bors r+

**Pull request merged**